iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -308,8 +315,11 @@ public async Task EnableEventReceiveAsync(TimeSpan timeout)
             {
                 if (_eventReceivingLink == null || _eventReceivingLink.IsClosing())
                 {
+                    _eventReceivingLink?.SafeClose();
+
                     _eventReceivingLink = await amqpIoTSession.OpenEventsReceiverLinkAsync(_deviceIdentity, timeout).ConfigureAwait(false);
-                    _eventReceivingLink.Closed += (obj, arg) => {
+                    _eventReceivingLink.Closed += (obj, arg) =>
```

Comment:
Another event to make sure we unhook from when _eventReceivingLink is disposed

Reply:
We should get together and talk about the overall approach for closing and re-initializing.